diagnostic_feedback/public/js/src/studio/studio_common.js

```diff
@@ -0,0 +1,1056 @@
+            };
+        }).get();
+    };
+
+    commonObj.getStep1Data = function () {
+        // Return first step data
+
+        var type = commonObj.getQuizType();
+        return {
+            title: $(quizTitle, element).val(),
+            description: $(quizDescription, element).val(),
+            type: type
+        }
+    };
+
+    commonObj.getStep2Data = function () {
+        // Get data of step2
+
+        var type = commonObj.getQuizType();
+        if (type === 'BFQ') {
+            return {
+                'categories': commonObj.getCategoriesList('category[name]')
+            }
+        } else {
+            return {
+                'ranges': commonObj.getRangesList('range[name]')
+            }
+        }
+
+    };
+
+    commonObj.getStep3Data = function () {
+        // Get step3 data before posting to server
+
+        var questionContainers = $(questionPanel + " " + questionDiv);
+        var questions = [];
+        $.each(questionContainers, function (i, container) {
+            var group = $(container).find(questionGroupField).val();
+
+            var questionObj = {
+                order: $(container).find(questionOrderField).val(),
+                question_title: $(container).find(questionTitleField).val(),
+                group: group !== "" ? group.trim() : initData.DEFAULT_GROUP,
+                question_txt: $(container).find(questionTxtField).val(),
+                choices: []
+            };
+
+            var id = $(container).find(questionIdField).first().val();
+            if (!id) {
+                id = commonObj.generateUniqueId();
+                $(container).find(questionIdField).first().val(id);
+            }
+            questionObj['id'] = id;
+            var answerChoicesInputs = $(container).find(choiceNameByClass);
+            $.each(answerChoicesInputs, function (j, choice) {
+                var answerChoice = {
+                    'choice_txt': $(choice).val(),
+                    'choice_value': $(choice).parent().nextAll(choiceValueByClass).first().val(),
+                    'choice_category': $(choice).parent().nextAll(choiceResult).val()
+                };
+                questionObj['choices'].push(answerChoice);
+            });
+            questions.push(questionObj);
+        });
+        return {'questions': questions};
+    };
+
+    commonObj.refreshAccordion = function (_id) {
+        // refresg an instance of jquery-accordion
+        $(_id).accordion('refresh');
+    };
+
+    commonObj.processCategories = function (categoriesContainer) {
+
+        //update attributes of all category fields after sort/delete
+        var remainingCategories = categoriesContainer.find(categoryDiv);
+
+        $.each(remainingCategories, function (i, category) {
+            var fields = $(category).find('input[type="text"], select, input[type="hidden"], textarea');
+
+            $(category).parent().prev().find(categoryOrder).html(i + 1);
```

Comment:
Nit: use ``.text`` for consistency even though there's no risk of XSS from a number. It just makes it easier to scan for safe usage, and it also means it will work if in the future the line is changed to include text as well as the number.